### PR TITLE
fixes 1773 added count_hosts method to puppetclass.rb

### PR DESCRIPTION
**Pull request by isratrade**

### CODE REVIEW

**Comment by ohadlevy** [comment]
do you mind breaking the commits into seperate issues? (e.g one to fix 1773 and another to change habtm <-> has_many  through)?

it would be easier to generate an release notes later on, e.g.

hostgroup class assignments is now audited.

thanks!

File: app/models/puppetclass.rb [hunk 1]
```diff
@@ -169,4 +171,26 @@ def self.value_to_sql(operator, value)
     return "%#{value}%"
   end
 
+  def count_hosts
+    #get number of hosts in host_classes for cases where host goes not belong to hostgroup
+    cnt1 =  self.hosts.length
+
+    if hostgroup_ids.empty?
+      cnt1
```
Comment (skottler) [inline app/models/puppetclass.rb]: Can you name this variable differently - cnt1 and cnt2 are distinct so I think their names should more accurately represent the values they store.

File: app/models/puppetclass.rb [hunk 2]
```diff
@@ -169,4 +171,26 @@ def self.value_to_sql(operator, value)
...
+      #get hostgroup_ids for puppetclass
+      array_hostgroup_ids = self.hostgroup_ids
+
+      #add hostgroup idss that are descendents of hostgroups for puppetclass
+      hostgroups.each do |hostgroup|
+        array_hostgroup_ids << hostgroup.descendant_ids
+      end
+
+      #get hosts part of hostgroup_ids
+      cnt2 =  Host.where(:hostgroup_id => array_hostgroup_ids).length
```
Comment (skottler) [inline app/models/puppetclass.rb]: Ditto, re: the comment above.

**Comment by skottler** [comment]
Also, can you write a test or two for the count_hosts method?

**Comment by ohadlevy** [comment]
@isratrade please recreate the commit, and ideally break it down to seperate pull requests... 
thanks

**Pull request closed without merging**